server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -46,29 +63,73 @@ public String typeName() {
             return CONTENT_TYPE;
         }
 
+        @Override
+        public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
+            return new StoredValueFetcher(context.lookup(), NAME);
+        }
+
         @Override
         public Query existsQuery(SearchExecutionContext context) {
-            // This query is not performance sensitive, it only helps assess
-            // quality of the data, so we may use a slow query. It shouldn't
-            // be too slow in practice since the number of unique terms in this
-            // field is bounded by the number of fields in the mappings.
             return new TermRangeQuery(name(), null, null, true, true);
         }
 
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
+            throw new IllegalArgumentException(
+                "aggregations on the '"
+                    + typeName()
+                    + "' field are supported for indices created by version "
+                    + IndexVersions.DOC_VALUES_FOR_IGNORED_META_FIELD
+                    + " or higher"
```

Comment:
I think we may want to avoid exposing version info in this exception. Perhaps the standard error for this would be fine, maybe there is no need to override this method?

Reply:
Note that the standard error prints something that you commented on before like

```
"Fielddata is not supported on field [_ignored] of type [_ignored]"
```